solidity/random-beacon/test/Groups.test.ts

```diff
@@ -652,7 +652,7 @@ describe("Groups", () => {
         const groupPublicKey1 = groupPublicKey
         const groupPublicKey2 = groupPublicKey
 
-        context("when both groups are candidate", async () => {
+        context("when the first group is active", async () => {
```

Comment:
I find the test cases in this context highly confusing. We are testing `activateCandidateGroup` and the scenario I believe we are dealing with is:
- group A added as candidate
- group A challenged
- group B added as candidate with the same pub key as A
- group B activated

The fact B overwrites A is checked in `addCandidateGroup` test, at line 408: `with the same group public key.
The fact we cannot add another candidate group when one with the same pub key is active is checked in `addCandidateGroup` test, at line 318: `with the same group public key`.

So here, I think what we want to check is if activation of B is possible. Something like:

when two groups are registered -> with the same group public key -> when the first group was challenged and replaced -> it should not be possible to activate the second group

Reply:
So you want to test:
* add group A as a candidate
* pop group A
* call `activateCandidateGroup` (it fails)
?

Reply:
- Add group A as candidate
- Pop group A
- Add group B as candidate (with the same pub key)
- Call activate on B and ensure it works

I think it's the only valid path for "when two groups are registered -> with the same group public key"

Reply:
Done in https://github.com/keep-network/keep-core/commit/fc955c3c1376e50df055de8ba33b8906c04b8e1b.

I basically used this flow and checked if it succeeded:
```
              await groups.addCandidateGroup(
                groupPublicKey1,
                members1,
                noMisbehaved
              )

              await groups.popCandidateGroup()

              await groups.addCandidateGroup(
                groupPublicKey2,
                members2,
                noMisbehaved
              )

              tx = await groups.activateCandidateGroup()             
```